models/demos/tg/llama3_70b/tests/test_llama_demo_nightly.py

```diff
@@ -0,0 +1,127 @@
+    check_device_mesh,
+)
+from models.demos.tg.llama3_70b.demo.demo import main, construct_arg
+
+
+@pytest.mark.timeout(240000)
+@skip_for_grayskull("Requires eth connected devices to run")
+@pytest.mark.parametrize(
+    "cluster_shape, device_mesh", [pytest.param((4, 8), (8, 4), id="4x8_grid")], indirect=["device_mesh"]
+)
+@pytest.mark.parametrize(
+    "llama_version",
+    (("llama3-tg"),),
+)
+@pytest.mark.parametrize(
+    "chat, prompts_file",
+    ((False, "models/demos/t3000/llama2_70b/demo/data/multi_prompt.json"),),
+    ids=("text_completion",),
+)
+@pytest.mark.parametrize("decode_only", (True,), ids=("decode_only",))
+@pytest.mark.parametrize("num_layers", (80,), ids=("80L",))
+@pytest.mark.parametrize(
+    "implementation, skip_model_load, n_devices",
+    (
+        (
+            "tt",
+            False,
+            8,
+        ),
+    ),
+    ids=("tt-70b-glx",),
+)
+@pytest.mark.parametrize(
+    "max_output_tokens, output_at_end, top_p, top_k, temperature",
+    ((128, True, 1, 1, 1.0),),
+    ids=("greedy",),
+)
+@pytest.mark.parametrize(
+    "ground_truth",
+    (None,),
+    ids=("check_disabled",),
+)
+@pytest.mark.parametrize(
+    "max_batch_size, max_context_len",
+    ((32, 2048),),
+    ids=("short_context",),
+)
+def test_llama3_tg_nightly_demo(
+    # model args
+    implementation,
+    skip_model_load,
+    num_layers,
+    # Generation args
+    max_output_tokens,
+    prompts_file,
+    output_at_end,
+    top_p,
+    top_k,
+    temperature,
+    chat,
+    # TT args
+    device_mesh,
+    cluster_shape,
+    n_devices,
+    decode_only,
+    llama_version,
+    ground_truth,
+    max_batch_size,
+    max_context_len,
+    use_program_cache,
+):
+    logger.info("Running LlamaModel demo")
+
+    ## Get model config
+    model_config, ckpt_dir, tokenizer_path, cache_path = setup_llama_env(
+        llama_version=llama_version,
+    )
+
+    check_device_mesh(device_mesh, model_config)
+
+    # TODO: Renable when issue #11089 is resolved
+    for i in device_mesh.get_device_ids():
+        device = device_mesh.get_device(i)
+        device.enable_async(True)
+
+    args = construct_arg(
+        implementation=implementation,
+        llama_version=llama_version,
+        ckpt_dir=ckpt_dir,
+        tokenizer_path=tokenizer_path,
+        skip_model_load=skip_model_load,
+        num_layers=num_layers,
+        max_batch_size=max_batch_size,
+        max_kv_context_len=max_context_len,
+        max_output_tokens=max_output_tokens,
+        prompts_file=prompts_file,
+        output_at_end=output_at_end,
+        top_p=top_p,
+        top_k=top_k,
+        temperature=temperature,
+        chat=chat,
+        device_mesh=device_mesh,
+        cluster_shape=cluster_shape,
+        n_devices=n_devices,
+        cache_path=cache_path,
+        decode_only=decode_only,
+        ground_truth=ground_truth,
+        print_output_as_generated=False,
+        print_output_at_end=True,
+    )
+    main(args)
```

Comment:
strange to import a function called `main`. Can we rename this to something more descriptive?

Reply:
Yes, renamed to run_demo